src/models/edge.ts

```diff
@@ -400,15 +402,56 @@ abstract class Edge<N extends INodeBase, E extends IEdgeBase> extends Subject im
   }
 
   setState(state: number): void;
+  setState(state: IGraphObjectStateParameters): void;
   setState(callback: (edge: IEdge<N, E>) => number): void;
-  setState(arg: number | ((edge: IEdge<N, E>) => number)): void {
+  setState(callback: (edge: IEdge<N, E>) => IGraphObjectStateParameters): void;
+  setState(
+    arg:
+      | number
+      | IGraphObjectStateParameters
+      | ((edge: IEdge<N, E>) => number)
+      | ((edge: IEdge<N, E>) => IGraphObjectStateParameters),
+  ): void {
+    let result: number | IGraphObjectStateParameters;
+
     if (isFunction(arg)) {
-      this._state = (arg as (edge: IEdge<N, E>) => number)(this);
+      result = (arg as (edge: IEdge<N, E>) => number | IGraphObjectStateParameters)(this);
     } else {
-      this._state = arg as number;
+      result = arg;
     }
+
+    if (typeof result === 'number') {
+      this._state = result;
+    } else if (typeof result === 'object') {
+      const options = result.options;
```

Comment:
For these checks, please use `utils/type.utils.ts` because of:
- cleaner code, `isNumber` vs `typeof x === 'number'
- handles edge cases, `isPlainObject` vs `typeof result === 'object'` (edge case is: `typeof null` will return `'object'`).